config/Config-images.in

```diff
@@ -153,6 +153,38 @@ menu "Target Images"
 		help
 		  Build a squashfs-lzma root filesystem.
 
+		config TARGET_SQUASHFS_COMPRESSION_OVERRIDE
+			bool "Override compression"
+			default n
+			depends on TARGET_ROOTFS_SQUASHFS
+			help
+			  Override default compression method
+
+		choice
+			prompt "Compression"
+			default TARGET_SQUASHFS_COMPRESSION_GZIP
+			depends on TARGET_ROOTFS_SQUASHFS
+			depends on TARGET_SQUASHFS_COMPRESSION_OVERRIDE
+			help
+			  Select squashfs compression.
+
+			config TARGET_SQUASHFS_COMPRESSION_GZIP
+				bool "gzip"
+
+			config TARGET_SQUASHFS_COMPRESSION_LZ4
+				bool "lz4"
+
+			config TARGET_SQUASHFS_COMPRESSION_LZO
+				bool "lzo"
+
+			config TARGET_SQUASHFS_COMPRESSION_XZ
+				bool "xz"
```

Comment:
You have to activate the compression standards also in the kernel configuration. Currently most of them are deactivated:
``
$ git grep SQUASHFS -- target/linux/generic/config-5.15
target/linux/generic/config-5.15:CONFIG_MTD_SPLIT_SQUASHFS_ROOT=y
target/linux/generic/config-5.15:CONFIG_SQUASHFS=y
target/linux/generic/config-5.15:# CONFIG_SQUASHFS_4K_DEVBLK_SIZE is not set
target/linux/generic/config-5.15:# CONFIG_SQUASHFS_DECOMP_MULTI is not set
target/linux/generic/config-5.15:CONFIG_SQUASHFS_DECOMP_MULTI_PERCPU=y
target/linux/generic/config-5.15:# CONFIG_SQUASHFS_DECOMP_SINGLE is not set
target/linux/generic/config-5.15:CONFIG_SQUASHFS_EMBEDDED=y
target/linux/generic/config-5.15:# CONFIG_SQUASHFS_FILE_CACHE is not set
target/linux/generic/config-5.15:CONFIG_SQUASHFS_FILE_DIRECT=y
target/linux/generic/config-5.15:CONFIG_SQUASHFS_FRAGMENT_CACHE_SIZE=3
target/linux/generic/config-5.15:# CONFIG_SQUASHFS_LZ4 is not set
target/linux/generic/config-5.15:# CONFIG_SQUASHFS_LZO is not set
target/linux/generic/config-5.15:# CONFIG_SQUASHFS_XATTR is not set
target/linux/generic/config-5.15:CONFIG_SQUASHFS_XZ=y
target/linux/generic/config-5.15:# CONFIG_SQUASHFS_ZLIB is not set
target/linux/generic/config-5.15:# CONFIG_SQUASHFS_ZSTD is not set
``

Reply:
Again, all much cleaner in the code nobody can see yet